File: iOS动画(持续添加中)/滚到导航栏的动画/HJNavigationAnimation/HJNavigationAnimation/ViewController.m
//
//  ViewController.m
//  HJNavigationAnimation
//
//  Created by coco on 16/6/23.
//  Copyright © 2016年 XHJ. All rights reserved.
//

#import "ViewController.h"
#import <Masonry.h>

@interface ViewController ()<UITableViewDataSource, UITableViewDelegate>
/**tableView*/
@property (nonatomic, strong)UITableView *tableView;
/**图像*/
@property (nonatomic, strong)UIImageView *imageV;
@end

@implementation ViewController
- (UIImageView *)imageV {
    if (!_imageV) {
        _imageV = [[UIImageView alloc] initWithFrame:CGRectMake(0, -250, self.view.frame.size.width, 250)];
        _imageV.contentMode = UIViewContentModeScaleAspectFit;
        _imageV.image = [UIImage imageNamed:@"城市"];
//        _imageV.backgroundColor = [UIColor redColor];
    }
    return _imageV;
}
- (UITableView *)tableView {
    if (!_tableView) {
        _tableView = ({
            UITableView *object = [[UITableView alloc] initWithFrame:CGRectZero style:(UITableViewStyleGrouped)];
            object.delegate = self;
            object.dataSource = self;
            object.contentInset = UIEdgeInsetsMake(250, 0, 0, 0);
            object;
        });
    }
    return _tableView;
}
- (void)viewDidLoad {
    [super viewDidLoad];
//    self.edgesForExtendedLayout = UIRectEdgeNone;
    [self configSubview];
}
- (void)configSubview {
    [self.view addSubview:self.tableView];
    [self.tableView addSubview:self.imageV];
    [self.tableView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.right.bottom.equalTo(self.view);
        make.top.mas_equalTo(250);
    }];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 10;
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:(UITableViewCellStyleValue1) reuseIdentifier:@"cell"];
    }
    cell.textLabel.text = indexPath.description;
    return cell;
}


@end

File: HJLinkLabel/HJLinkLabel/ViewController.swift
//
//  ViewController.swift
//  HJLinkLabel
//
//  Created by coco on 16/9/23.
//  Copyright © 2016年 XHJ. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        createLabel()
        
    }

    func createLabel() {
        let label = HJLinkLabel()
        label.numberOfLines = 0
        label.frame = CGRect(x: 50, y: 200, width: 500, height: 30)
        label.attributedText = NSAttributedString(string: "我是大傻瓜, 你是大傻逼", attributes: [NSForegroundColorAttributeName: UIColor.grayColor(), NSFontAttributeName: UIFont.systemFontOfSize(17)])
        label.addLinkString("我") { (title) in
            print("我")
        }
        label.addLinkString("你是", attribute: [NSForegroundColorAttributeName: UIColor.redColor()]) { (title) in
            print("你是")
        }
        label.numberOfLines = 0
        self.view.addSubview(label
        )
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

